runtime/signals.c

```diff
@@ -183,7 +188,7 @@ CAMLexport void caml_leave_blocking_section(void)
      [signals_are_pending] is 0 but the signal needs to be
      handled at this point. */
   signals_are_pending = 1;
-  caml_raise_if_exception(caml_process_pending_signals_exn());
+  //caml_raise_if_exception(caml_process_pending_signals_exn());
```

Comment:
I think it is usually a bug to check for pending signals in `caml_leave_blocking_section`? You generally have a result (success or error) of whatever was in the blocking section, and discarding it because a signal handler raises is probably not what you want.

Reply:
I agree with @jhjourdan.  Please remove the commented-out code.  Optionally, you could add a comment that explains "We used to process pending signals at this point but it is not a good idea because...".